### PR TITLE
Fix: unset GIT_DIR and GIT_WORK_TREE for git operations #7958

### LINKED ISSUE #7958: `pod repo update` should unset GIT_DIR before running git operations
* [X] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I updated my Podfile and ran `pod update someDep` from within a `git rebase` operation, which triggered a `pod repo update`.

## What did you expect to happen?

This should have updated the cocoapods repo.

## What happened instead?

Because I was in the middle of a `git rebase`, the `GIT_DIR` environment variable was set. This made CocoaPods's `git fetch` operation actually apply the operation to my local repo instead of the cocoapods spec repo.

## Notes

The `GIT_DIR` environment variable should be unset, but there's probably others that need to be unset too.

### PR DESCRIPTION
**Pull request by tripleCC**
closes #7958

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix: unset GIT_DIR and GIT_WORK_TREE for git operations #7958  
  [tripleCC](https://github.com/tripleCC)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Fix: unset GIT_DIR and GIT_WORK_TREE for git operations #7958
```
Comment (dnkoutso) [inline CHANGELOG.md]: can remove #7958 here.

**Comment by dnkoutso** [comment]
@tripleCC what about all other `git!` invocations?

**Comment by tripleCC** [comment]
@dnkoutso  git! and git is defined by `extend Executable` and invoking `executable` dynamically in many classes

should I clear GIT_DIR and GIT_WORK_TREE env in executable method when name is git ?

What do you think is the most appropriate approach to correct it ?

**Comment by segiddins** [comment]
I think this is actually fine

**Comment by dnkoutso** [comment]
Needs a rebase and we can land.

**Comment by dnkoutso** [comment]
ping @tripleCC for rebase

**Pull request merged**